addons/web/static/tests/webclient/actions/concurrency_tests.js

```diff
@@ -695,6 +695,32 @@ QUnit.module("ActionManager", (hooks) => {
         assert.containsOnce(webClient, ".o_form_view");
     });
 
+    QUnit.test(
+        "dialog will only open once for two rapid actions with the target new",
+        async function (assert) {
+            assert.expect(3)
+            const def = makeDeferred();
+            const mockRPC = async (route, args) => {
+                if (args.method === "onchange") {
+                   return def;
+                }
+            };
+
+            const webClient = await createWebClient({ serverData, mockRPC });
+            doAction(webClient, 5);
+            await nextTick();
+            assert.containsNone(webClient, ".o_form_view");
+
+            doAction(webClient, 25);
+            await nextTick();
+            assert.containsNone(webClient, ".o_form_view");
+
+            def.resolve();
+            await nextTick();
+            assert.containsOnce(webClient, ".o_form_view", "dialog should open only once");
```

Comment:
```suggestion
            assert.containsOnce(webClient, ".o_dialog .o_form_view", "dialog should open only once");
```
Because we must ensure that the executed action is target="new"